pkg/api/serve.go

```diff
@@ -116,7 +116,11 @@ func Serve(
 		r.Mount("/oidc/login", NewOIDCLoginPageHandler(oidcConfig, sessionStore, oauthConfig, logger))
 	}
 	r.Mount("/logout", NewLogoutHandler(sessionStore, logger, cfg.GetAuthLogoutRedirectURL()))
-	r.Mount("/", NewUIHandler(gatewayDomains, snippets))
+
+	// Disable serving of embedded static assets when using an external client
+	if cfg.GetUIEnabled() {
+		r.Mount("/", NewUIHandler(gatewayDomains, snippets))
+	}
```

Comment:
1. We need to handle the default route in case the request arrives to one of the gateway domains - trace how the call uses `gatewayDomains`
2. Suggestion - configuration parameter 'ui.disable' which will make it easy on the defaults - as for `false` we don't need to add code to set the default and nothing to keep in memory by default.

Reply:
1. Fixed to accommodate this case
2. `ui.enabled` was used for consistency with other config structs